demo/src/main/java/com/novoda/movies/rateable/RateableMovieViewHolder.java

```diff
@@ -0,0 +1,120 @@
+package com.novoda.movies.rateable;
+
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.RatingBar;
+import android.widget.TextView;
+
+import com.novoda.accessibility.AccessibilityServices;
+import com.novoda.accessibility.Action;
+import com.novoda.accessibility.Actions;
+import com.novoda.accessibility.ActionsAccessibilityDelegate;
+import com.novoda.accessibility.ActionsAlertDialogCreator;
+import com.novoda.movies.R;
+
+import java.util.Arrays;
+
+class RateableMovieViewHolder extends RecyclerView.ViewHolder {
+
+    static RateableMovieViewHolder inflate(ViewGroup parent) {
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.item_view_rateable_movie, parent, false);
+        return new RateableMovieViewHolder(view);
+    }
+
+    private final ImageView posterImageView;
+    private final TextView titleTextView;
+    private final ImageView likeImageView;
+    private final RatingBar ratingBar;
+    private final AccessibilityServices a11yServices;
+
+    RateableMovieViewHolder(View itemView) {
+        super(itemView);
+        posterImageView = itemView.findViewById(R.id.item_rateable_image_poster);
+        titleTextView = itemView.findViewById(R.id.item_rateable_text_title);
+        likeImageView = itemView.findViewById(R.id.item_rateable_image_like);
+        ratingBar = itemView.findViewById(R.id.item_rateable_rating);
+        a11yServices = AccessibilityServices.newInstance(itemView.getContext());
+    }
+
+    void bind(RateableMovieViewModel viewModel) {
+        posterImageView.setImageResource(viewModel.poster);
+        titleTextView.setText(viewModel.title);
+        likeImageView.setImageResource(viewModel.liked ? R.drawable.ic_favorite_24dp : R.drawable.ic_favorite_border_24dp);
+        ratingBar.setRating(viewModel.rating);
+        Actions actions = collateActionsFor(viewModel);
+        ActionsAccessibilityDelegate a11yDelegate = new ActionsAccessibilityDelegate(itemView.getResources(), actions);
+        ViewCompat.setAccessibilityDelegate(itemView, a11yDelegate);
+        itemView.setContentDescription(viewModel.title + ", rating " + viewModel.rating + ", liked: " + viewModel.liked);
+
+        if (a11yServices.isSpokenFeedbackEnabled() || a11yServices.isSwitchAccessEnabled() || !itemView.isInTouchMode()) {
```

Comment:
this if/else will change the behaviour of our view though. Here, we check:

- if an accessibility service that uses spoken feedback is enabled (like TalkBack or Select to Speak)
- if Switch Access (an accessibility service that allows users to navigate and interact with their device with a one or two button input) is enabled
- or if the user is navigating with a keyboard/dpad (`!view.isInTouchMode()`)

for any of these cases, we'll consider that the user is not accessing the views directly with touch, and delegate to the `bindForIndirectAccess(actions)` method. the `setClickLabel` thing is so the accessibility service knows what will happen if the user clicks on the view: in this case, they'll click to "see more options".

in the other case (the `else`), they'll click the item view and it'll trigger the `viewModel.actions.onSelectMovie()` directly.